lib/SIL/SILVerifier.cpp

```diff
@@ -4345,6 +4345,24 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     }
     
     SILInstructionVisitor::visitSILBasicBlock(BB);
+
+    // Visit the instructions in the BB, to find holes in the lexical scopes.
+    // We use a fairly simple heuristic here (Explain heuristic).
+    llvm::DenseSet<const SILDebugScope *> AlreadySeenScopes;
+    if (BB->empty())
+      return;
+    SILInstruction &First = *BB->begin();
+    const SILDebugScope *LastSeenScope = First.getDebugScope();
+    AlreadySeenScopes.insert(LastSeenScope);
+    for (SILInstruction &SI : *BB) {
+      auto *DS = SI.getDebugScope();
+      if (!AlreadySeenScopes.count(DS)) {
```

Comment:
nit: if (AlreadySeenScopes.insert(DS).second)?

Reply:
Yes, definitely. I'll change it :)